be/src/util/thrift_util.cpp

```diff
@@ -149,4 +149,24 @@ bool t_network_address_comparator(const TNetworkAddress& a, const TNetworkAddres
 
     return false;
 }
+
+void thrift_from_json_string(::apache::thrift::TBase* base, const std::string& json_val) {
+    using namespace apache::thrift::transport;
+    using namespace apache::thrift::protocol;
+    auto* buffer = new TMemoryBuffer((uint8_t*)json_val.c_str(), (uint32_t)json_val.size());
+    std::shared_ptr<TTransport> trans(buffer);
+    TJSONProtocol protocol(trans);
+    base->read(&protocol);
+}
+
+const std::string thrift_to_json_string(const ::apache::thrift::TBase* base) {
```

Comment:
appears that thrift already provides this helper function: https://github.com/apache/thrift/blob/0.13.0/lib/cpp/src/thrift/protocol/TJSONProtocol.h#L308

`template <typename ThriftStruct> std::string ThriftJSONString(const ThriftStruct& ts);`

Reply:
it's really difficult to find it. I guess it's better to add a impl here.

Reply:
It's up to you to make the final call whether need to add these two new interfaces. I am ok with this.

W.r.t. "symmetric",
```
template <typename ThriftStruct> std::string ThriftJSONString(const ThriftStruct& ts);
template <typename ThriftStruct> void JSONStringThrift(ThriftStruct *ts, const std::string &jsonStr);
```
looks symmetric too. :)

Reply:
I don't like template if we can use normal functions when there is no need to pursue performance.